### PR TITLE
Quarantine `TestUnixDomainSocketWithUrl`

### PR DESCRIPTION
**Pull request by dougbu**
- see #28067
- add guards with detailed messages to help debug future failures

### CODE REVIEW

**Comment by dougbu** [comment]
Left over from my ops rotation&hellip;

**Comment by dougbu** [comment]
And, no, I don't really know who owns the Libuv functional tests 😺

File: src/Servers/Kestrel/test/FunctionalTests/UnixDomainSocketsTests.cs [hunk 1]
```diff
@@ -129,6 +129,7 @@ async Task EchoServer(ConnectionContext connection)
 
 #if LIBUV
         [OSSkipCondition(OperatingSystems.Windows, SkipReason = "Libuv does not support unix domain sockets on Windows.")]
+        [QuarantinedTest("https://github.com/dotnet/aspnetcore/issues/24608")]
```
Comment (Tratcher) [inline src/Servers/Kestrel/test/FunctionalTests/UnixDomainSocketsTests.cs:132]: Wrong issue number?

Comment (dougbu) [reply src/Servers/Kestrel/test/FunctionalTests/UnixDomainSocketsTests.cs:132]: Yup

**Review by Tratcher — APPROVED** [review]

**Pull request merged**